connector/src/main/scala/org/apache/spark/sql/cassandra/DefaultSource.scala

```diff
@@ -27,93 +32,46 @@ import org.apache.spark.sql.streaming.OutputMode
-   *    keyspace     -- keyspace name, required
-   *    cluster      -- cluster name, optional, default name is "default"
-   *    pushdown     -- true/false, optional, default is true
-   *    Cassandra connection settings  -- optional, e.g. spark.cassandra.connection.timeoutMS
-   *    Cassandra Read Settings        -- optional, e.g. spark.cassandra.input.fetch.sizeInRows
-   *    Cassandra Write settings       -- optional, e.g. spark.cassandra.output.consistency.level
-   *
-   * When push_down is true, some filters are pushed down to CQL.
-   *
-   */
-  override def createRelation(
-    sqlContext: SQLContext,
-    parameters: Map[String, String]): BaseRelation = {
-
-    val (tableRef, options) = TableRefAndOptions(parameters)
-    CassandraSourceRelation(tableRef, sqlContext, options, None)
-  }
+class DefaultSource() extends TableProvider with DataSourceRegister {
 
-  /**
-   * Creates a new relation for a cassandra table given table, keyspace, cluster and push_down
-   * as parameters and explicitly pass schema [[StructType]] as a parameter
-   */
-  override def createRelation(
-    sqlContext: SQLContext,
-    parameters: Map[String, String],
-    schema: StructType): BaseRelation = {
-
-    val (tableRef, options) = TableRefAndOptions(parameters)
-    CassandraSourceRelation(tableRef, sqlContext, options, Some(schema))
+  override def getTable(schema: StructType, partitioning: Array[Transform], properties:java.util.Map[String,String]): Table = {
+    getTable(new CaseInsensitiveStringMap(properties))
   }
 
-  /**
-   * Creates a new relation for a cassandra table given table, keyspace, cluster, push_down and schema
-   * as parameters. It saves the data to the Cassandra table depends on [[SaveMode]]
-   */
-  override def createRelation(
-    sqlContext: SQLContext,
-    mode: SaveMode,
-    parameters: Map[String, String],
-    data: DataFrame): BaseRelation = {
-
-    val (tableRef, options) = TableRefAndOptions(parameters)
-    val table = CassandraSourceRelation(tableRef, sqlContext, options, None)
-
-    mode match {
-      case Append => table.insert(data, overwrite = false)
-      case Overwrite => table.insert(data, overwrite = true)
-      case ErrorIfExists =>
-        if (table.buildScan().isEmpty()) {
-          table.insert(data, overwrite = false)
-        } else {
-          throw new UnsupportedOperationException(
-            s"""'SaveMode is set to ErrorIfExists and Table
-               |${tableRef.keyspace + "." + tableRef.table} already exists and contains data.
-               |Perhaps you meant to set the DataFrame write mode to Append?
-               |Example: df.write.format.options.mode(SaveMode.Append).save()" '""".stripMargin)
-        }
-      case Ignore =>
-        if (table.buildScan().isEmpty()) {
-          table.insert(data, overwrite = false)
-        }
-    }
-
-    CassandraSourceRelation(tableRef, sqlContext, options, None)
+  override def shortName(): String = "cassandra"
```

Comment:
DOC NOTE: "cassandra" is valid format, can't believe we didn't turn this on when it was added (Spark 1.5! )